lib/apytypes/_array_functions.py

```diff
@@ -28,16 +28,49 @@ def squeeze(a, axis=None):
     return squeeze(axis=axis)
 
 
+def convolve(a, v, mode="full"):
+    """
+    Return the discrete linear convolution of two one-dimensional arrays.
+
+    Parameters
+    ----------
+    a : :class:`APyFloatArray` or :class:`APyFixedArray`
+        First one-dimensional array of length :code:`M`
+
+    v : :class:`APyFloatArray` or :class:`APyFixedArray`
+        Second one-dimensional array of length :code:`N`.
+
+    mode : {'full', 'same', 'valid'}, default: 'full'
+        'full':
+            Returns the full convolution for each point of overlap. The resulting
+            single-dimensional shape will have length :code:`N + M - 1`. Boundary
+            effects occurs for points where the `a` and `v` do not overlap completely.
+        'same':
+            Returns a convolution of length :code:`max(M, N)`. Boundary effects still
+            occur around the edges of the result.
+        'valid':
+            Returns the convoltuion for each point of full overlap. The resulting
+            single-dimensional shape will have length :code:`max(M, N) - min(M, N) + 1`
+    """
+    try:
+        return a.convolve(v, mode=mode)
+    except AttributeError:
+        raise TypeError("Can only perform convoltuion of same-type apytypes arrays")
```

Comment:
I take it that this is not the final error?

Reply:
It was my idea of the exception message. What do you have in mind?

Reply:
Why would "same types" be mentioned here? It will not be triggered when different types are used? Just seems a bit strange to state it like that.

Reply:
Okay, I thought L56 would throw an `AttributeError`, but it looks like a `TypeError` is thrown if `a` and `v` are not of the same type. Will fix.

I'll add a more helpful error message as well.